### ISSUE TITLE
FP S2589 (`no-gratuitous-expressions`): inlined react expression

### ISSUE THREAD

**Issue by vilchik-elena** [SonarSource/SonarJS#2807]
In the following example `children` is reported inside JSX. 
```js
if (children) {
  return (
    <div>
      {right && children}
    </div>
  );
}
```

Reported in the internal thread https://discuss.sonarsource.com/t/reactjs-print-statement-incorrectly-detected-as-a-simple-boolean-expression/8524

**Comment by vilchik-elena** [comment]
Internal link to FP https://peach.sonarsource.com/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&issues=AXrLLt99JDEs3N5jdT0q&open=AXrLLt99JDEs3N5jdT0q
